src/Microsoft.HttpRepl.IntegrationTests/Mocks/MockHttpMessageHandler.cs

```diff
@@ -0,0 +1,24 @@
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.HttpRepl.IntegrationTests.Mocks
+{
+    public class MockHttpMessageHandler : HttpMessageHandler
+    {
+        private HttpResponseMessage response;
+
+        public MockHttpMessageHandler(HttpResponseMessage response)
```

Comment:
A more flexible mock (which can be done in the future) would be to make this handler allow you to register a URL (or HttpRequestMessage if you want to allow separate HttpMethods, etc) with a response.  E.g. something like

```
var mockHttpMessageHandler = new MockHttpMessageHandler().Register("url", cannedResponse)
                                                         .Register("url2", alternateResponse);
new HttpClient(mockHttpMessageHandler);
```

Reply:
Sure, thanks!

I don't see a use case for the current set of tests in this pr. Let me know if I am missing something. The one you suggested might be super useful for RunCommand. Let me know if you had something else in mind.

Reply:
What do you guys think about doing something like this?

https://github.com/xunit/xunit/issues/188#issuecomment-480822091